app/controllers/api/databases.php

```diff
@@ -147,6 +150,92 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
     return $attribute;
 }
 
+
+App::init()
+    ->groups(['timeout'])
+    ->inject('request')
+    ->inject('dbForProject')
+    ->action(function (Request $request, Database $dbForProject) {
+        $key = md5(json_encode([
+            $request->getMethod(),
+            $request->getURI(), // Contains databaseId & collectionId
+            $request->getParam('queries')
+        ]));
+        /** @var Document $document */
+        $document = Authorization::skip(fn() => $dbForProject->getDocument('slowQueries', $key));
+        if ($document->getAttribute('blocked') === true) {
+            throw new Exception(Exception::TIMEOUT_BLOCKED);
+        }
+    });
+
+App::error()
+    ->groups(['timeout'])
+    ->inject('error')
+    ->inject('request')
+    ->inject('dbForProject')
+    ->action(function (Throwable $error, Request $request, Database $dbForProject) {
+        try {
+            if ($error instanceof Timeout) {
+                $route = Request::getRoute();
+                $collectionId = $route->getParamValue('collectionId');
+                $databaseId = $route->getParamValue('databaseId');
+                $queries = $request->getParam('queries', []);
+
+                $queriesValidator = new ArrayList(new Text(APP_LIMIT_ARRAY_ELEMENT_SIZE), APP_LIMIT_ARRAY_PARAMS_SIZE);
+                if (!$queriesValidator->isValid($queries)) {
+                    throw new Exception(Exception::GENERAL_SERVER_ERROR);
+                }
+
+                $uidValidator = new UID();
+                if (!$uidValidator->isValid($databaseId)) {
+                    throw new Exception(Exception::GENERAL_SERVER_ERROR);
+                }
+
+                if (!$uidValidator->isValid($collectionId)) {
+                    throw new Exception(Exception::GENERAL_SERVER_ERROR);
+                }
+
+                $key = md5(json_encode([
+                    $request->getMethod(),
+                    $request->getURI(), // Contains databaseId & collectionId
+                    $request->getParam('queries')
+                ]));
+
+                /** @var Document $document */
+                $document = Authorization::skip(fn() => $dbForProject->getDocument('slowQueries', $key));
+                if ($document->isEmpty()) {
+                    $document = Authorization::skip(fn()=>$dbForProject->createDocument('slowQueries', new Document([
+                        '$id' => $key,
+                        'blocked' => false,
+                        'count' => 1,
+                        'queries' => $queries,
+                        'databaseId' => $databaseId,
+                        'collectionId' => $collectionId,
+                        'path' => $request->getURI()
+                ])));
+                } else {
+                    $document->setAttribute('count', $document->getAttribute('count') + 1);
+                    $max = intval(App::getEnv('_APP_SLOW_QUERIES_MAX_HITS', 9999));
```

Comment:
Let's set the default max to the same as the `.env` file

Reply:
The value we have now is for tests, it will be different than the production,
This is for a case the timeouts are 'Enabled' but from some reason _APP_SLOW_QUERIES_MAX_HITS is empty and we want a big number fallback